### PR TITLE
Upgrade dependencies and fix lint complaints

### PR DESCRIPTION
**Pull request by AlexMSmithCA**
Resolves the following complaints:

```
$ eslint . --ignore-path .gitignore
--
  | Done in 3.90s.
  |  
  | /fusion-release/packages/fusionjs/fusion-plugin-i18n-react/singleton.js
  | 3:1  error  Flow file annotation is missing  flowtype/require-valid-file-annotation
  |  
  | /fusion-release/packages/fusionjs/fusion-plugin-i18n-react/src/with-translations.js
  | 7:1  error  Flow file annotation is missing  flowtype/require-valid-file-annotation
  |  
  | ✖ 2 problems (2 errors, 0 warnings)
  |  
```

### CODE REVIEW

File: src/with-translations.js [hunk 1]
```diff
@@ -2,9 +2,11 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
  */
 
-import React, {Component} from 'react';
+import * as React from 'react';
```
Comment (KevinGrandon) [inline src/with-translations.js]: I think we had some issues with this syntax before causing things to not build correctly. It also doesn't look like you need specific types from React, so wondering if we could revert this or explicitly import types as needed?

Comment (AlexMSmithCA) [reply src/with-translations.js]: Hm, interesting.  It is the [recommended approach by Flow](https://flow.org/en/docs/react/types/).

We use the type for {{ComponentType}} on [line 23](https://github.com/fusionjs/fusion-plugin-i18n-react/pull/134/files#diff-6671dfa49d21c6bd618ccc6408d585b4R23) so I will import it explicitly instead.

Comment (KevinGrandon) [reply src/with-translations.js]: But I guess we do this do this in other modules now, so maybe it's fine? /shrug.

**Comment by AlexMSmithCA** [comment]
!merge

**Pull request merged**